apps/wallet/src/engine/runners/solana/collectibles.ts

```diff
@@ -28,29 +28,38 @@ export type GenericNft = Nft | Sft | SftWithToken | NftWithToken;
 type SolanaCollectibleDocument = NftDocument<SolanaCollectible>;
 type SolanaCollectionDocument = CollectionDocument<SolanaCollection>;
 
-export const getCollectiblesOnChain = async (
+export const queryCollectibles = async (
 	connection: Connection,
 	cluster: NetworkCluster,
 	wallet: PublicKey,
-): Promise<NftDocument<SolanaCollectible>[]> => {
+) => {
 	const mpl = new Metaplex(connection);
 	const rawNfts = await throttle(() => {
 		return mpl.nfts().findAllByOwner({ owner: wallet });
 	})();
 
-	const nfts = await Promise.all(
+	return await Promise.all(
 		rawNfts.map(async (metadata) => {
 			try {
 				const nft = await loadCollectibleMetadata(mpl, metadata, wallet);
 
-				return constructCollectibleDocument(wallet.toString(), nft, cluster);
+				const collectible = constructCollectibleDocument(
```

Comment:
it is ok to throw error when a single NFT have problem?

Reply:
it is in try catch block